src/ui/Button.tsx

```diff
@@ -262,91 +262,126 @@ type ButtonProps = Omit<InlineProps, 'size'> & {
   active?: boolean;
 };
 
-const Button = ({
-  iconName,
-  suffix,
-  variant,
-  disabled,
-  loading,
-  children,
-  customChildren,
-  shouldHideText,
-  onClick,
-  className,
-  title,
-  size,
-  forwardedAs,
-  type,
-  active,
-  ...props
-}: ButtonProps) => {
-  const isBootstrapVariant = (
-    Object.values(BootStrapVariants) as string[]
-  ).includes(variant);
-  const isLinkVariant = variant === ButtonVariants.LINK;
-
-  if (variant === ButtonVariants.SECONDARY) variant = 'secondary';
-
-  const BaseButtonWithForwardedAs = (props) => (
-    <BaseButton {...props} forwardedAs={forwardedAs} />
-  );
-
-  const forwardedButton = forwardedAs ? BaseButtonWithForwardedAs : BaseButton;
-  const bootstrapProps = isBootstrapVariant
-    ? { forwardedAs: forwardedButton, variant }
-    : {};
-
-  const propsToApply = {
-    ...bootstrapProps,
-    disabled,
-    onClick,
-    className,
-    title,
-    size,
-    type,
-    active,
-    ...getInlineProps(props),
-  };
-
-  const clonedSuffix = suffix
-    ? // @ts-expect-error
-      cloneElement(suffix, {
-        // @ts-expect-error
-        ...suffix.props,
-        css: `align-self: flex-end`,
-        key: 'suffix',
-      })
-    : undefined;
-
-  const inner = loading ? (
-    <Spinner
-      className="button-spinner"
-      variant={SpinnerVariants.LIGHT}
-      size={SpinnerSizes.SMALL}
-    />
-  ) : (
-    [
-      iconName && <Icon name={iconName} key="icon" />,
-      customChildren
-        ? children
-        : !shouldHideText && (
-            <Text flex={1} textAlign="left" key="text">
-              {children}
-            </Text>
-          ),
-      clonedSuffix,
-    ]
-  );
-
-  if (isBootstrapVariant) {
-    return (
-      <BootstrapButton {...propsToApply} css={customCSS}>
-        {inner}
-      </BootstrapButton>
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      iconName,
+      suffix,
+      variant,
+      disabled,
+      loading,
+      children,
+      customChildren,
+      shouldHideText,
+      onClick,
+      className,
+      title,
+      size,
+      forwardedAs,
+      type,
+      active,
+      ...props
+    },
+    ref,
+  ) => {
+    const isBootstrapVariant = (
+      Object.values(BootStrapVariants) as string[]
+    ).includes(variant);
+    const isLinkVariant = variant === ButtonVariants.LINK;
+
+    if (variant === ButtonVariants.SECONDARY) variant = 'secondary';
```

Comment:
Not sure I get this one, doesn't it reassign the same value to the variable than it has then? 🤔 I know it was like that before but feels weird

Reply:
It also seems to me that this isn't needed